Content.Server/Speech/EntitySystems/SpanishAccentSystem.cs

```diff
@@ -36,21 +36,29 @@ private string InsertS(string message)
             return msg;
         }
 
-        private string ReplaceQuestionMark(string message)
+        private string ReplacePunctuation(string message)
         {
             var sentences = AccentSystem.SentenceRegex.Split(message);
             var msg = "";
             foreach (var s in sentences)
             {
-                if (s.EndsWith("?", StringComparison.Ordinal)) // We've got a question => add ¿ to the beginning
+                var toInsert = "";
```

Comment:
This should just use stringbuilder as it's going to allocate a lot of strings for this.

Reply:
Added StringBuilder to that one and the older code in there too.